core/services/headreporter/telemetry_reporter.go

```diff
@@ -0,0 +1,53 @@
+package headreporter
+
+import (
+	"context"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/services/telemetry"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+	"math/big"
+)
+
+type (
+	telemetryReporter struct {
+		logger    logger.Logger
+		endpoints map[*big.Int]commontypes.MonitoringEndpoint
+	}
+)
+
+func NewTelemetryReporter(chainContainer legacyevm.LegacyChainContainer, lggr logger.Logger, monitoringEndpointGen telemetry.MonitoringEndpointGenerator) *telemetryReporter {
+	endpoints := make(map[*big.Int]commontypes.MonitoringEndpoint)
+	for _, chain := range chainContainer.Slice() {
+		endpoints[chain.ID()] = monitoringEndpointGen.GenMonitoringEndpoint("EVM", chain.ID().String(), "", synchronization.HeadReport)
```

Comment:
Would we also need some form of check if telemetry is enabled in cfg? What do we see in the plugins? I assume this would fail if not

Reply:
done